scipy/io/mmio.py

```diff
@@ -553,18 +649,16 @@ def _write(self, stream, a, comment='', field=None, precision=None):
             else:
                 raise TypeError('unexpected dtype kind ' + kind)
 
-        if rep == self.FORMAT_ARRAY:
-            symm = self._get_symmetry(a)
-        else:
-            symm = self.SYMMETRY_GENERAL
+        if symmetry is None:
+            symmetry = self._get_symmetry(a)
 
         # validate rep, field, and symmetry
         self.__class__._validate_format(rep)
         self.__class__._validate_field(field)
-        self.__class__._validate_symmetry(symm)
+        self.__class__._validate_symmetry(symmetry)
 
         # write initial header line
-        stream.write(asbytes('%%%%MatrixMarket matrix %s %s %s\n' % (rep,field,symm)))
+        stream.write(asbytes('%%%%MatrixMarket matrix %s %s %s\n' % (rep,field,symmetry)))
```

Comment:
I believe PEP8 will complain about the lack of commas between the arguments in the tuple.

Also, might read better as:

```
stream.write(asbytes('%%MatrixMarket matrix {0} {1} {2]\n'.format(rep, field, symmetry))
```